lib/fizzy/cxx20/bit.hpp

```diff
@@ -24,3 +25,45 @@ template <class To, class From>
     return dst;
 }
 }  // namespace fizzy
+
+#if __cplusplus > 201703L
+
+#include <bit>
+
+namespace fizzy
+{
+constexpr uint32_t countl_zero(uint32_t value) noexcept
+{
+    // C++ standard outputs 32 when value is 0.
+    return std::countl_zero(value);
+}
+
+constexpr uint64_t countl_zero(uint64_t value) noexcept
+{
+    // C++ standard outputs 64 when value is 0.
+    return std::countl_zero(value);
+}
+}  // namespace fizzy
+
+#else
+
+namespace fizzy
+{
+inline uint32_t countl_zero(uint32_t value) noexcept
+{
+    // NOTE: Wasm specifies this case, but C/C++ intrinsic leaves it as undefined.
```

Comment:
Here we are not copying Wasm behaviour, but C++20 behaviour. These functions must 100% match how `countl_zero` behaves in C++20.

Reply:
In the original execute.cpp versions, the return type was uint32_t and uint64_t for these bit counting functions. https://github.com/wasmx/fizzy/blob/master/lib/fizzy/execute.cpp#L260

But c++ 20 standard library and c++ intrinsics both have them returning int. Which version should I do?